pkg/operator/firstteamcommentcontroller/firstteamcomment_controller.go

```diff
@@ -0,0 +1,138 @@
+package firstteamcommentcontroller
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/eparis/bugzilla"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	errutil "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog"
+
+	"github.com/mfojtik/bugzilla-operator/pkg/cache"
+	"github.com/mfojtik/bugzilla-operator/pkg/operator/config"
+	"github.com/mfojtik/bugzilla-operator/pkg/slack"
+)
+
+type FirstTeamCommentController struct {
+	config            config.OperatorConfig
+	newBugzillaClient func() cache.BugzillaClient
+	slackClient       slack.ChannelClient
+}
+
+func NewFirstTeamCommentController(operatorConfig config.OperatorConfig, newBugzillaClient func() cache.BugzillaClient, slackClient slack.ChannelClient, recorder events.Recorder) factory.Controller {
+	c := &FirstTeamCommentController{
+		config:            operatorConfig,
+		newBugzillaClient: newBugzillaClient,
+		slackClient:       slackClient,
+	}
+	return factory.New().WithSync(c.sync).ResyncEvery(2*time.Hour).ToController("FirstTeamCommentController", recorder)
+}
+
+func (c *FirstTeamCommentController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	client := c.newBugzillaClient()
+	var errors []error
+
+	for name, comp := range c.config.Components {
+		if !comp.AssignFirstDeveloperCommentor {
+			continue
+		}
+		if comp.Lead == "" {
+			continue
+		}
+
+		nonLeads := config.ExpandGroups(c.config.Groups, comp.Developers...)
+		nonLeads = nonLeads.Delete(comp.Lead)
+
+		query := fmt.Sprintf(
+			"email1=%s&email2=%s&emailassigned_to2=1&emaillongdesc1=1&emaillongdesc3=1&emailtype1=regexp&emailtype2=equals",
+			url.QueryEscape(strings.Join(nonLeads.List(), "|")),
+			url.QueryEscape(comp.Lead),
+		)
+		klog.Warning(query)
+		leadAssignedBugs, err := client.Search(bugzilla.Query{
+			Product:   []string{"OpenShift Container Platform"},
+			Status:    []string{"NEW"},
+			Component: []string{name},
+			Raw:       query,
+		})
+		if err != nil {
+			syncCtx.Recorder().Warningf("BuglistFailed", err.Error())
+			continue
+		}
+		klog.Infof("%d NEW bugs found assigned to lead", len(leadAssignedBugs))
+
+	nextBug:
+		for _, b := range leadAssignedBugs {
+			comments, err := client.GetCachedBugComments(b.ID, b.LastChangeTime)
+			if err != nil {
+				syncCtx.Recorder().Warningf("GetCachedBugComments", err.Error())
+				continue
+			}
+
+			var firstTeamCommentor string
+			for _, c := range comments {
+				creator := c.Creator
+				if !strings.ContainsRune(creator, '@') {
+					creator = creator + "@redhat.com"
+				}
+				if strings.Contains(c.Text, "LifecycleStale") {
+					continue
+				}
+				if nonLeads.Has(creator) && firstTeamCommentor == "" && b.Creator != creator {
```

Comment:
on line 49 you removed the lead from nonLeads, so the lead can never be the 'firstTeamCommentor'    is this going to steal all bugs, not just those they've never commented on? I guess we'd see that quickly in the logs if that were the case.

Reply:
The lead is the one who gets all bugs assigned automatically. This loop tries to find those where another team member commented, but didn't self-assign. So if the lead ever commented, he payed attention to the bug and this mechanism of auto-assignment is disabled. Makes sense?

Reply:
Overall goal here: avoid that we have tons of bugs which are NEW, but commented.